p2p/src/authenticated/discovery/actors/listener.rs

```diff
@@ -127,14 +156,8 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Network + Rng + CryptoRng + Metri
             .expect("failed to bind listener");
 
         // Loop over incoming connections as fast as our rate limiter allows
+        let mut limited = 0;
         loop {
-            // Ensure we don't attempt to perform too many handshakes at once
```

Comment:
Instead of limiting the number of connections we'll accept per unit of time, we limit the concurrency. This reduces the likelihood we build up some backlog of incoming connections that are all stale by the time we get to them (we can rush through those).